starknet_py/utils/crypto/facade.py

```diff
@@ -1,19 +1,45 @@
+import functools
 import os
-from typing import Optional
+from typing import Optional, Sequence, Callable
 
 from crypto_cpp_py.cpp_bindings import (
     cpp_hash,
     get_cpp_lib_file,
     ECSignature,
 )
-from starkware.cairo.common.hash_state import compute_hash_on_elements
 from starkware.cairo.lang.vm.crypto import pedersen_hash as default_hash
 from starkware.crypto.signature.signature import sign
 
 from starknet_py.net.client_models import Call
 
-# PREFIX_TRANSACTION = encoded 'StarkNet Transaction'
-PREFIX_TRANSACTION = 476441609247967894954472788179128007176248455022
+
+# Interface
+def use_cpp_variant() -> bool:
+    force_disable_ext = (
+        os.getenv("DISABLE_CRYPTO_C_EXTENSION", "false").lower() == "true"
+    )
+    cpp_lib_file = get_cpp_lib_file()
+    return not force_disable_ext and bool(cpp_lib_file)
+
+
+def pedersen_hash(left: int, right: int) -> int:
+    if use_cpp_variant():
+        return cpp_hash(left, right)
+    return default_hash(left, right)
+
+
+def compute_hash_on_elements(
```

Comment:
So we're leaving it in the facade.py after all?

Reply:
Yup, `compute_hash_on_elements` would be here. Previously, we talked about compute_address

Reply:
Okay, makes sense